hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java

```diff
@@ -86,7 +86,15 @@ protected boolean isCaseInsensitive(JdbcEnvironment jdbcEnvironment) {
 	}
 
 	private boolean isUnderscoreRequired(final char before, final char current, final char after) {
-		return Character.isLowerCase( before ) && Character.isUpperCase( current ) && Character.isLowerCase( after );
+		if (Character.isDigit(current)) {
+			if (Character.isDigit(before)) {
+				return false;
+			}
+			if (Character.isLowerCase(before)) {
+				return true;
+			}
```

Comment:
So `hello1` -> `hello_1`? I don't think I see why that's correct.

It seems to me that we should simply be treating digits exactly as we treat lowercase characters, no?